src/DIRAC/WorkloadManagementSystem/Service/JobStateUpdateHandler.py

```diff
@@ -77,13 +75,15 @@ def export_updateJobFromStager(cls, jobID, status):
             status = result["Value"]["Status"]
             if status == JobStatus.STAGING:
                 if i:
-                    infoStr = "Found job in Staging after %d seconds" % i
+                    infoStr = f"Found job in Staging after {i} seconds"
                 break
             time.sleep(1)
         if status != JobStatus.STAGING:
-            return S_OK("Job is not in Staging after %d seconds" % trials)
+            return S_OK(f"Job is not in Staging after {trials} seconds")
 
-        result = cls.__setJobStatus(int(jobID), status=jobStatus, minorStatus=minorStatus, source="StagerSystem")
+        result = JobStatusUtility(cls.jobDB, cls.jobLoggingDB, cls.elasticJobParametersDB).setJobStatus(
```

Comment:
JobStatusUtility(cls.jobDB, cls.jobLoggingDB, cls.elasticJobParametersDB) can be made a class attribute in the initializeHandler() to avoid its instantiation each time it is used - a small overhead but still...